lib/bloc/user_profile/currency_data.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:convert';
+
+Map<String, CurrencyData> currencyDataFromJson(String str) =>
+    new Map.from(json.decode(str)).map((k, v) => new MapEntry<String, CurrencyData>(k, CurrencyData.fromJson(k, v)));
+
+class CurrencyData {
```

Comment:
CurrencyData class should be part of the Currency service and not under the user_profile directory.